src/diffusers/utils/hub_utils.py

```diff
@@ -129,3 +134,116 @@ def create_model_card(args, model_name):
 
     card_path = os.path.join(args.output_dir, "README.md")
     model_card.save(card_path)
+
+
+def extract_commit_hash(resolved_file: Optional[str], commit_hash: Optional[str] = None):
+    """
+    Extracts the commit hash from a resolved filename toward a cache file.
+    """
+    if resolved_file is None or commit_hash is not None:
+        return commit_hash
+    resolved_file = str(Path(resolved_file).as_posix())
+    search = re.search(r"snapshots/([^/]+)/", resolved_file)
+    if search is None:
+        return None
+    commit_hash = search.groups()[0]
+    return commit_hash if REGEX_COMMIT_HASH.match(commit_hash) else None
+
+
+def try_cache_hub_download(
+    repo_id: str,
+    filename: str,
+    *args,
+    cache_dir: Union[str, Path, None] = None,
+    subfolder: Union[str, Path, None] = None,
+    _commit_hash: Optional[str] = None,
+    **kwargs,
+) -> Union[os.PathLike, str]:
+    """Wrapper method around hf_hub_download:
+    https://huggingface.co/docs/huggingface_hub/main/en/package_reference/file_download#huggingface_hub.hf_hub_download
+    that first tries to load from cache before pinging the Hub"""
+    if _commit_hash is not None:
+        # If the file is cached under that commit hash, we return it directly.
+        resolved_file = try_to_load_from_cache(
+            repo_id, filename, cache_dir=cache_dir, subfolder=subfolder, revision=_commit_hash
+        )
+        if resolved_file is not None:
+            if resolved_file is not _CACHED_NO_EXIST:
+                return resolved_file
+            else:
+                raise EnvironmentError(f"Could not locate {filename} inside {repo_id}.")
+
+    return hf_hub_download(repo_id, filename, *args, cache_dir=cache_dir, subfolder=subfolder, **kwargs)
```

Comment:
Here if you already know the revision because `_commit_hash` is provided, you can just pass it directly to `hf_hub_download`. Just for clarification about `hf_hub_download`:
- if `revision` is not a commit hash (i.e., either `main`, a specific branch/ref or a tag): a HEAD call is made to get the commit hash + same process as below 
- if `revision` is a commit hash:
    - if the corresponding snapshot already exists:
        - if the requested file is already cached, return it (no HTTP calls has been made)
        - if the requested file is not cached, cache it and return (only 1 GET call made)
    - if the corresponding snapshot is missing:
        - download and cache the file, and return it (only 1 GET call made)

Here what we want to do is to avoid the HEAD call because we already know which commit hash is associated. If you pass it as revision, you don't need all the logic to `try_to_load_from_cache` as it's already built in `hf_hub_download`.

I made the suggestion below (:arrow_down:) as an example but I think it's better to directly call `hf_hub_download` with the commit hash as `revision` and completely remove `try_cache_hub_download` (as @williamberman suggested I think?)

```suggestion
    if _commit_hash is not None:
        kwargs["revision"] = _commit_hash
    return hf_hub_download(repo_id, filename, *args, cache_dir=cache_dir, subfolder=subfolder, **kwargs)
```

Reply:
yes deleted everything - thanks!